tests/submodule/add.c

```diff
@@ -128,3 +128,51 @@ void test_submodule_add__url_relative_to_workdir(void)
 
 	assert_submodule_url("TestGitRepository", git_repository_workdir(g_repo));
 }
+
+void test_submodule_add__path_exists_in_index(void)
+{
+	git_index *index;
+	git_submodule *sm;
+	git_buf dirname = GIT_BUF_INIT;
+	git_buf filename = GIT_BUF_INIT;
+
+	/* In this repo, HEAD (master) has no remote tracking branc h*/
+	g_repo = cl_git_sandbox_init("testrepo");
+
+	cl_git_pass(git_buf_joinpath(&dirname, git_repository_workdir(g_repo), "TestGitRepository"));
```

Comment:
I'd rename "TestGitRepository" to e.g. just "subdirectory" or "directory". Otherwise it's a bit confusing, as that path should not represent a real git repository but should only contain a file later on.